nx/lib/nx.ex

```diff
@@ -6408,15 +6408,160 @@ defmodule Nx do
 
   """
   @doc type: :ndim
-  def dot(t1, axes1, t2, axes2) do
+  def dot(t1, axes1, t2, axes2), do: dot(t1, axes1, [], t2, axes2, [])
```

Comment:
We will probably eventually work this out, but it's okay for now so it doesn't break anything and we can handle it in a later PR. Just documenting for now :)